src/ecash/service/mod.rs

```diff
@@ -0,0 +1,216 @@
+use crate::ecash;
+
+use std::collections::HashSet;
+use std::convert::TryInto;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::net::{Ipv4Addr, SocketAddr};
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use tracing::{debug, error, warn};
+
+mod handlers;
+
+#[derive(Clone)]
+struct MintState {
+    ln: ln_rs::Ln,
+    mint: Arc<tokio::sync::Mutex<cdk::mint::Mint>>,
+}
+
+pub async fn mint_service(
+    config_file_arg: String,
+) -> anyhow::Result<tokio::task::JoinHandle<()>> {
+    let ecash_config = match ecash::config::EcashConfig::new(&Some(config_file_arg)) {
+        Ok(mint_settings) => mint_settings,
+        Err(e) => {
+            error!("{}. Halting.", e);
+            std::process::exit(1);
+        }
+    };
+
+    // Create db_path parent directory if it doesn't exist
+    guarantee_parent_path(&ecash_config.mint.db_path)?;
+    let localstore = cdk::mint::RedbLocalStore::new(
+        ecash_config
+            .mint
+            .db_path
+            .to_str()
+            .expect("PathBuf always converts to &str"),
+    )?;
+
+    let mint = cdk::mint::Mint::new(
+        Arc::new(localstore),                                       // localstore
+        bip39::Mnemonic::from_str(&ecash_config.mint.mnemonic)?,    // mnemonic
+        HashSet::new(),                                             // keysets_info
+        cdk::amount::Amount::ZERO,                                  // min_fee_reserve
+        0.0,                                                        // percent_fee_reserve
+    )
+    .await?;
+
+    debug!("Mint created");
+
+    let cln_socket = expand_path(
+        ecash_config
+            .ln
+            .cln_path
+            .clone()
+            .ok_or(anyhow::anyhow!("cln socket not defined"))?
```

Comment:
`use anyhow::Context;`

```suggestion
            .context("cln socket not defined")?
```
and so on...

Reply:
98b7ad102b4551eee2fa013caecb3c07013b6dc5